java/bulk-export/bulk-export-core/src/main/java/sleeper/bulkexport/model/BulkExportQuery.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.bulkexport.model;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * A request for a bulk export on a given table.
+ */
+public class BulkExportQuery {
+    private final String exportId;
+    private final String tableId;
+    private final String tableName;
+
+    private BulkExportQuery(Builder builder) {
+        exportId = builder.exportId != null ? builder.exportId : UUID.randomUUID().toString();
+
+        if (builder.tableId == null && builder.tableName == null) {
+            throw new BulkExportQueryValidationException(exportId, "tableId or tableName field must be provided");
+        } else if (builder.tableId != null && builder.tableName != null) {
+            throw new BulkExportQueryValidationException(exportId,
+                    "tableId or tableName field must be provided, not both");
```

Comment:
This is passing the randomised export ID into the exception, but the exception checks if the export ID is null. Should we pass in the export ID from the builder instead?

Reply:
Updated to use the builder exportId